lab-nassir/route/cat-router.js

```diff
@@ -0,0 +1,69 @@
+'use strict';
+
+const Router = require('express').Router;
+const debug = require('debug')('cat:cat-router');
+const createError = require('http-errors');
+const jsonParser = require('body-parser').json();
+
+const Cafe = require('../model/cafe');
+const Cat = require('../model/cat');
+
+const catRouter = module.exports = new Router();
+
+catRouter.post('/api/cafe/:cafeId/cat', jsonParser, function(req, res, next){
+  debug('Hit POST router');
+  Cafe.findByIdAndAddCat(req.params.cafeId, req.body)
+  .then(cat => res.json(cat))
+  .catch(next);
+});
+
+catRouter.get('/api/cafe/:cafeId/cat/:catId', function(req, res, next){
```

Comment:
It would be simpler if you just required the client to pass in the catId and found the cat without going through the cafe. There's no need to reference the cafe to find a cat :)